godot-core/src/builtin/array.rs

```diff
@@ -566,8 +566,26 @@ impl<T: VariantMetadata + ToVariant> Array<T> {
 //     ...
 // }
 
-impl<T: VariantMetadata> GodotFfi for Array<T> {
-    ffi_methods! { type sys::GDExtensionTypePtr = *mut Opaque; .. }
+unsafe impl<T: VariantMetadata> GodotFfi for Array<T> {
+    ffi_methods! { type sys::GDExtensionTypePtr = *mut Opaque;
+        fn from_sys;
+        fn sys;
+        fn from_sys_init;
+        // SAFETY:
+        // Nothing special needs to be done beyond a `std::mem::swap` when returning an Array.
+        fn move_return_ptr;
+    }
+
+    // SAFETY:
+    // Arrays are properly initialized through a `from_sys` call, but the ref-count should be
+    // incremented as that is the callee's responsibility.
+    //
+    // Using `std::mem::forget(array.share())` increments the ref count.
```

Comment:
this is me documenting that the implementation of `from_arg_ptr` upholds the safety invariants that `GodotFfi` requires from an implementation of `from_arg_ptr`. I could move it inside if you want.

Reply:
In that case, it should probably be part of the `unsafe impl` block docs, see comment above.

It's a bit non-local, but you could make two paragraphs in the `impl` docs, one for each function. The code is still relatively close, so that one sees the `impl` docs when looking at the function.